promise.js

```diff
@@ -330,6 +362,21 @@ class PromisePool extends EventEmitter {
     });
   }
 
+  transaction(options,promiseCallback) {
+    return this.getConnection()
+    .then(con => 
+      con.transaction(options,promiseCallback)
+      .then(res => {
+        con.release();
+        return res;
+      })
+      .catch(err => {
```

Comment:
Nit, you probably want to merge the `then` and `catch` into one `then(ok, err)` chain instead, otherwise the catch handler would potentially catch `.release()` errors.